src/main/scala/no/ndla/articleapi/service/converters/HTMLCleaner.scala

```diff
@@ -149,10 +149,17 @@ trait HTMLCleaner {
       val minimumIngressWordCount = 3
       val firstSection = Option(el.select("body>section").first)
       val firstDivSection = Option(el.select("body>section:eq(0)>div").first)
-
-      val ingress = firstSection.flatMap(getIngressText) match {
-        case None => firstDivSection.flatMap(getIngressText)
-        case ing => ing
+      val secondDivSection = Option(el.select("body>section:eq(0)>div:eq(0)>div").first)
+
+      // Look for ingress according to the following priorities:
+      //   1. first paragraph in first section, ei. <section><p> HERE </p></section>
+      //   2. first paragraph in first div inside first section, ei. <section><div><p> HERE </p></div></section>
+      //   3. first paragraph in first nested div inside first section, ei. <section><div><div><p> HERE </p></div></div></section>
+      val ingress = (firstSection.flatMap(getIngressText), firstDivSection, secondDivSection) match {
+        case (Some(ing), _, _) => Some(ing)
+        case (None, _, Some(secondDiv)) => getIngressText(secondDiv)
```

Comment:
Det er fordi `secondDiv` her er et `Element` og ikke en `Option[Element]` (`secondDiv` blir destructured i `case`n). `firstSection` er en `Option[Element]` og `getIngressText` returnerer en `Option[List[String]]`, så for at resultatet skal bli en `Option[List[String]]` må man bruke `flatMap` (ekvivalent med `map(..).flatten`)